### ISSUE TITLE
Improve cli

### ISSUE THREAD

**Issue by anbraten** [woodpecker-ci/woodpecker#3386]
The woodpecker cli could get some polishing:

- [x] updater #3382 
  - [x] background update checker
  - [x] update command with `--force` option
- [ ] interactive setup #3384 
  - [x] setup command 
  - [ ] command to enter server-url on first start 
  - [x] improved auth (opens browser with UI, send back token to localhost via POST) 
- [ ] follow step logs when step is still running
- [ ] improve error handling for missing args (missing repo id / name results in an parseInt error atm)
- [ ] improve outputs
  - [ ]  use some table layouts
- [ ] watch options
  - [ ] agents list
  - [ ] agent tasks
  - [ ] pipeline status
- [ ] improve exec command
  - [ ] support multiple workflow files
  - [ ] exec with metadata from UI #1155 
- [ ] detect repo from `.git/config` in pwd #910

**Comment by xoxys** [comment]
@woodpecker-ci/maintainers I would like to work on `improve error handling for missing args (missing repo id / name results in an parseInt error atm)` How do we want to handle it? Do we really want to continue using args? IMO, it would be better to drop args completely and just use flags. Any reason to keep using args?

**Comment by anbraten** [comment]
I think its quite common practice to use args (see kubectl, docker, ...) therefore I would suggest to stick to it and add a empty-string check with a more descriptive error in front of the `parseInt`.

**Comment by xoxys** [comment]
Obviously, but I could also name a lot of tools without args. We could have a way more clean code without arg validation, as flag validation and required flags is a built-in feature of urfave/cli. However, if args is preferred, that works for me as well.

**Comment by lafriks** [comment]
I don't know if urfave/cli still checks that required arguments are not empty if required (you can provide argument but provide it with empty value) so check for empty value could still be needed

**Comment by xoxys** [comment]
There is no required args feature, just a required flags option.

**Comment by jinnatar** [comment]
While the new setup flow seems nice, it utterly fails on a headless machine. As a quick fix I'd recommend echoing the URL that it's trying to apparently open in a browser or otherwise provide a flow where I can open the URL on another device and feed back a token on the cli.

**Comment by 6543** [comment]
yes we should check if the env has an wayland or xorg session, and if not we print out this sugestion and fail as we run "headless" ...

**Comment by 6543** [comment]
I have a dedicated `cli exec tui` summary issue: https://github.com/woodpecker-ci/woodpecker/issues/6488

where you can track what the current state & plan is :)
